text/0000-buildpack-registry-search-api.md

```diff
@@ -0,0 +1,172 @@
+# Meta
+[meta]: #meta
+- Name: Buildpack Registry Search API
+- Start Date: 2020-11-11
+- Author(s): elbandito
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+This is a proposal for a service API that enables clients to search for buildpacks.
+
+# Definitions
+[definitions]: #definitions
+
+**API** - Application Programmer Interface
+
+**Buildpack Registry** - the CNB centeralized [registry](https://github.com/buildpacks/registry-index) that contains metadata for all the community supported buildpacks 
+
+**Service** - an independent web process running in the cloud that provides RESTful endpoints
+
+**Client** - represents a physical end-user or software interface e.g. CLI, web UI, etc.
+ 
+Make a list of the definitions that may be useful for those reviewing. Include phrases and words that buildpack authors or other interested parties may not be familiar with.
+
+# Motivation
+[motivation]: #motivation
+
+As the CNB project makes progress towards supporting a centralized Buildpack Registry, I anticipate the need to quickly search for and identify specific buildpacks.  To support this, I propose the creation of an API search service that customers could use directly and/or indirectly via a web dashboard or CLI.
+
+- What use cases does it support?  
+  - Customers looking to quickly indentify usefulful buildpacks to build their applications.
+- What is the expected outcome?  
+  - Customers will be able to quickly find relevant buildpacks for their use cases.
+
+# What it is
+[what-it-is]: #what-it-is
+
+An external API service that exposes endpoints for retrieving buildpack metadata.  Initially, I envision this service providing a few GET endpoints to support plain text query searches, and individual buildpack data retrieval. 
+
+### Supported Endpoints
+
+- **GET /buildpacks**
```

Comment:
We aren't going to be big enough where this will be on the db, but I've noticed that npm did [deprecate their `./-/all` endpoint](https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md#get-v1search).

If we are going to do this, do we need to look into pagination as well?

Reply:
We should support pagination for large responses from search queries, but I agree that the `GET /buildpacks` probably doesn't make sense.